sprint_17

Gvozdkov opened the pull request:
Мой основной проект по не известной мне причине выдал: - " The project ‘Tracker’ is damaged and cannot be opened due to a parse error. Examine the project file for invalid edits or unresolved source control conflicts." =(
По этому, я создал новый проект и восстанавливал из старого. Надеюсь функциональность не пропала ^_^

Commit:
Есть скриншот-тесты для главного экрана.

Commit:
В проекте подключена библиотека AppMetrica через менеджер CocoaPods.
Действия на главной отправляют:
открытие,
закрытие,
тап на кнопке добавления трека,
тап на треке,
тап на кнопке фильтра,
выбор редактирования в контекстном меню,
выбор удаления в контекстном меню.

Commit:
В проекте подключена библиотека AppMetrica через менеджер CocoaPods.

Commit:
Настроена анимация выделения ячейки как в макете

LizaIlinykh commented on Tracker/AppDelegate.swift:
```diff
@@ -1,36 +1,41 @@
-//  AppDelegate.swift
-//  Tracker
-//
-//  Created by Алексей Гвоздков on 14.03.2024.
```
не принято удалять сгенерированную шапку файла

LizaIlinykh commented on Tracker/Helpers/ColorsForTheTheme.swift:
```diff
@@ -1,8 +1,77 @@
 
-import Foundation
+final class ColorsForTheTheme {
+    static let shared = ColorsForTheTheme()
```
принято делать инициализатор приватным при использовании синглтона, это позволяет избежать случайного создания других экземпляров 
https://vikramios.medium.com/swift-singleton-design-pattern-2cdbd52516a8

LizaIlinykh commented on Tracker/UI/Tracker/CreatingTracker/NewTracker/Category/NewCategoryViewController.swift:
```diff
@@ -1,8 +1,134 @@
+extension NewCategoryViewController: UITextFieldDelegate {
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        if textField == nameTrackerTextField {
+            var colorButton = false
```
булевые переменные лучше так не называть, переменная с таким именем вводит в заблуждение, так как я думаю , что там хранится цвет

Commit:
Реализовано добавление экрана EditTrackerViewController для редактирования трекера. Добавлено переключение на данный экран при нажатии на ячейку и выборе опции 'Редактировать' в меню на экране TrackerViewController.

Commit:
Экран Онбординг соответствует макету

Commit:
Обновлены заглушки на экранах:
1) TrackerViewController
2) CategoryViewController

Commit:
Произведена локализация:
1) Экран FiltersViewController
2) Кнопка filtersButton на экране TrackerViewController

Commit:
Внесены следующие изменения:
1. По нажатию на ячейку с трекером появляется контекстное меню с пунктами:
- закрепить (закрепляет и открепляет ячейку)
- редактировать (запускает флоу с редактированием ячейки)
- удалить (удаляет ячейку с тренером)
2. Таблица с привычками теперь увеличивается в зависимости от количества привычек, а не фиксированного разрешения.
3. Добавлена метрика:
- edit для тапа на кнопку редактировать в контекстном меню
- delete для тапа на кнопку удалить в контекстном меню
